truss/util/log_utils.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import threading
+from typing import List, Optional
+
+
+class LogInterceptor(logging.Handler):
```

Comment:
[non-blocking] since this is used in contexts where people might be interacting programmatically, it might be good to test this with a couple different loggers just to be sure

Reply:
Do you mean testing with different formatters? Is the concern here just a general unittest for this or actually asserting that it works with diverse setups?